ipatests/test_xmlrpc/test_automember_plugin.py

```diff
@@ -275,6 +277,52 @@ def set_automember_process_modify_ops(value):
     run(cmd)
 
 
+def wait_automember_rebuild():
+    """Wait for an asynchronous automember rebuild to finish
+
+    Lookup the rebuild taskid and then loop until it is finished.
+
+    If no task is found assume that the task is already finished.
+    """
+    if not have_ldap2:
+        raise unittest.SkipTest('server plugin not available')
+    ldap = ldap2(api)
+    ldap.connect()
+
+    # give the task a chance to start
+    time.sleep(1)
+
+    try:
+        task_entries, unused = ldap.find_entries(
+            base_dn=REBUILD_TASK_CONTAINER,
+            filter='(&(!(nstaskexitcode=0))(scope=*))')
+    except errors.NotFound:
+        # either it's done already or it never started
+        return
+
+    # we run these serially so there should be only one at a time
+    assert len(task_entries) == 1
```

Comment:
In theory we could have more than one task with the filter !(nstaskexitcode=0), for instance one on-going task with no nstaskexitcode yet and a previous task that failed somehow (nstaskexitcode = 1). But I agree that this case has low probability during the tests and it would probably be overkill to retrieve the task id from automember-rebuild --nowait and look for this id in this method.